### PR TITLE
Refactor of try blocks, enumsets, scanner params and code defects.

### LINKED ISSUE #74: Change Try-Catch blocks to Try with Resources statements.
Coverity:
```
        1. new_resource: new java.io.FileWriter(location, true) creates a new resource.
   	2. var_assign: Assigning: writer = resource returned from new java.io.FileWriter(location, true).
 89                FileWriter writer = new FileWriter(location, true);
   	3. noescape: Resource writer is not closed or saved in append.
   	4. Throwing java.io.IOException (or subclass) from call to append; exiting method with uncaught exception.
```

Try changing try-catch in main.appendCustomerHistory() to a try with resources from **item 9 of Effective Java** to remove Coverity detected defect.

We may also want to revise this method out of our source code. There are other Try-Catch blocks that can be changes also.

**Create a 'try with resource refactor' branch and pull request to master when all try catch blocks are try with resources statements.**

### PR DESCRIPTION
**Pull request by JonathanGrant92**
Refactored to try with resources & enumsets, passed scanner as params, & fixed warnings. Resolves #74

### CODE REVIEW

File: src/main/java/com/dreamteam/database/OrderDatabase.java [hunk 1]
```diff
@@ -248,17 +215,15 @@ public void processOrders() {
 		String date = "2020-01-01";
 		ProductDatabase products = ProductDatabase.getProducts();
 
-		try {
+		try (Scanner order_scanner = new Scanner(new File(order_log_path))) {
```
Comment (JonathanGrant92) [inline src/main/java/com/dreamteam/database/OrderDatabase.java:218]: I should make this try with resource statement more readable!

**Review by kevvo15 — APPROVED** [review]

File: src/main/java/com/dreamteam/database/main.java [hunk 2]
```diff
@@ -201,15 +198,14 @@ public static void runMenu() {
...
+	 *
+	 * @param date
 	 */
-
 	public static void dailyAssetsReport(String date) {  //prints out report to console and txt file with assets in product_database and customer orders and sales in order_database
 		NumberFormat formatter = NumberFormat.getCurrencyInstance(); //to format the print statements in dollar form
-		String report_path = "files/reports/dailyreport_" + date + ".txt";
+		String report_path = "files/reports/daily-report_" + date + ".txt";
 		
 		try {
```
Comment (JonathanGrant92) [inline src/main/java/com/dreamteam/database/main.java:210]: This should probably be a try with resources to, but the file will need to be created via a file writer object.

**Comment by JonathanGrant92** [comment]
@kevvo15 I am finished pushing changes to this branch. It's good to go. Perhaps you can prompt someone else to also review the code before merge, like @alexandersanford since some of these changes may impact his branch.

**Review by alexandersanford — APPROVED** [review]
These are great improvements. Being able to stop resource leaks and provide our code with more built-in resource management is excellent. Great work!

**Review by domschneider — APPROVED** [review]

**Pull request merged**